tools/partition_algorithms/random.py

```diff
@@ -0,0 +1,39 @@
+import json
+import logging
+import sys
+import os
+import numpy as np
+
+from utils import setdir
+from utils import array_readwriter
+
+def _random_partition(metadata, num_parts):
+    num_nodes_per_type = [sum(_) for _ in metadata['num_nodes_per_chunk']]
+    ntypes = metadata['node_type']
+    for ntype, n in zip(ntypes, num_nodes_per_type):
+        logging.info('Generating partition for node type %s' % ntype)
+        parts = np.random.randint(0, num_parts, (n,))
+        array_readwriter.write(ntype + '.txt', parts, name='csv')
+
+def random_partition(metadata, num_parts, output_path):
+    """
+    Randomly partition the graph described in metadata and generate partition ID mapping
+    in :attr:`output_path`.
+
+    A directory will be created at :attr:`output_path` containing the partition ID
+    mapping files named "<node-type>.txt" (e.g. "author.txt", "paper.txt" and
+    "institution.txt" for OGB-MAG240M).  Each file contains one line per node representing
+    the partition ID the node belongs to.
+    """
+    with setdir(output_path):
+        _random_partition(metadata, num_parts)
+
+# Run with PYTHONPATH=${GIT_ROOT_DIR}/tools
+# where ${GIT_ROOT_DIR} is the directory to the DGL git repository.
+if __name__ == '__main__':
```

Comment:
Add a `argparser` and make it runnable like:
```bash
python /my/repo/dgl/tools/partition_algo/random.py
    --in-dir=/mydata/MAG240M-LSC_chunked/
    --out-dir=/mydata/MAG240M-LSC_2parts/
    --num-parts=2
```

Reply:
Done.  Although the first argument would be the metadata file path instead of the input directory.